src/configobject.h

```diff
@@ -34,23 +35,32 @@
 
 class ConfigKey {
   public:
-    ConfigKey();
+    ConfigKey(); // is required for qMetaTypeConstructHelper()
+    ConfigKey(const ConfigKey& key);
     ConfigKey(const QString& g, const QString& i);
-    ConfigKey(const char* g, const char* i);
     static ConfigKey parseCommaSeparated(QString key);
 
     inline bool isNull() const {
         return group.isNull() && item.isNull();
     }
 
+    // comparison function for ConfigKeys. Used by a QHash in ControlObject
+    inline bool operator==(const ConfigKey& other) const {
```

Comment:
Very good and concise explanation :)

On 01/12/2016 09:41 AM, RJ Ryan wrote:
>
> In src/configobject.h 
> <https://github.com/mixxxdj/mixxx/pull/501#discussion_r49427091>:
>
> >      static ConfigKey parseCommaSeparated(QString key);
> >
> >      inline bool isNull() const {
> >          return group.isNull() && item.isNull();
> >      }
> >
> > +    // comparison function for ConfigKeys. Used by a QHash in ControlObject
> > +    inline bool operator==(const ConfigKey& other) const {
>
> Our style guide gives a reason why to prefer it: "Prefer to define 
> non-modifying binary operators as non-member functions. If a binary 
> operator is defined as a class member, implicit conversions will apply to 
> the right-hand argument, but not the left-hand one. It will confuse your 
> users if a < b compiles but b < a doesn't."
>
> https://google.github.io/styleguide/cppguide.html#Operator_Overloading
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/501/files#r49427091>.
>